ssl_certificate.t

```diff
@@ -142,6 +142,10 @@ foreach my $name ('pass') {
 		. "-subj /CN=$name/ -out $d/$name.crt -key $d/$name.key "
 		. "-passin pass:pass >>$d/openssl.out 2>&1") == 0
 		or die "Can't create $name certificate: $!\n";
+	system('openssl x509 -addtrust serverAuth -trustout '
+		. "-in $d/$name.crt -out $d/$name-trust.crt "
+		. ">>$d/openssl.out 2>&1") == 0
+		or die "Can't create certificate with trust for $name: $!\n";
 }
 
 $t->write_file('password_file', 'pass');
```

Comment:
This results in misleading output in case of a caught regression:
```
1..13
ok 1 - default certificate
ok 2 - default context
ok 3 - virtual server certificate
ok 4 - virtual server context
ok 5 - certificate - no context
ok 6 - virtual server - no context
not ok 7 - ssl_password_file
#   Failed test 'ssl_password_file'
#   at ssl_certificate.t line 167.
#                   undef
#     doesn't match '(?^:password)'
ok 8 - session reused
ok 9 - session id context match
ok 10 - session id context distinct
ok 11 - no certificate
ok 12 - no alerts
ok 13 - no sanitizer errors
# Looks like you failed 1 test of 13.
```
Also, this file is restricted to run on OpenSSL 1.0.2.
As such, and as a general rule, it is better put these tests separately.

Also, the proposed patch won't catch regressions in loading CA certificates with trust aux.

Reply:
I agree. The separate tests are more informative. I've reviewed your proposed tests and updated this PR with them after fixing some cosmetic issues.